includes/wikia/api/ArticlesApiController.class.php

```diff
@@ -1048,6 +1056,86 @@ public function getAsJson() {
 		$this->setResponseData( $result, '', self::SIMPLE_JSON_VARNISH_CACHE_EXPIRATION );
 	}
 
+	/**
+	 * Get some or all of the article top level sections as an array
+	 *
+	 * Note: If sections 1, 2 and 3 are requested, but section 3 is a subsection, the returned
+	 * array will only have two items: section 1 & 2 (section 2 containing subsection 3).
+	 * All top level sections will contain all of their subsections, regardless of how many
+	 * were requested. So if sections 1, 2 and 3 are requested, and section 2 contains 3 subsections,
+	 * section two will be returned with all 3 subsections.
+	 *
+	 * @param mixed $sectionsToGet Array of section numbers (could need URL decoding) or the string "all"
+	 * @param ParserOutput $parsedArticle
+	 * @param Article $article
+	 * @param Title $title
+	 * @return array
+	 * @throws BadRequestApiException
+	 */
+	private function getArticleSections( $sectionsToGet, ParserOutput $parsedArticle, Article $article, $title ) {
+		$content = [];
+		$articleAsWikiText = $article->getContent();
+		$parsedSections = $parsedArticle->getSections();
+
+		// decode strings like "%5B1,2,3%5D" or "[1,2,3]"
+		if ( is_string( $sectionsToGet ) ) {
+			$sectionsToGet = json_decode( urldecode( $sectionsToGet ) );
+		}
+		// convert "all" or ["all"] to array of numbers
+		if ( $sectionsToGet === 'all' || in_array( 'all', $sectionsToGet ) ) {
+			$sectionsToGet = range( 0, count( $parsedArticle->getSections() ) );
+		}
+		// should be an array by this point
+		if ( is_array( $sectionsToGet ) ) {
+			foreach ( $sectionsToGet as $section ) {
+				// Only retrieve top level sections.
+				// $parsedSections doesn't include section 0, so subtract 1
+				$isTopLevel = $section > 0 && $parsedSections[$section - 1]['toclevel'] === 1;
+				if ( $section == 0 || $isTopLevel ) {
+					$content[] = $this->getArticleSection( $articleAsWikiText, $title, $section );
+				}
+			}
+		} else {
+			throw new BadRequestApiException( 'Sections must be an array of numbers or "all"' );
+		}
+		return $content;
+	}
+
+	/**
+	 * Get the HTML for an article section
+	 * @param string $articleAsWikiText
+	 * @param Title $title
+	 * @param integer $section
+	 * @return string
+	 * @todo Supposedly getSection works with HTML, but I could only get it to work with wikitext
+	 */
+	private function getArticleSection( $articleAsWikiText, $title, $section ) {
+		/** @var Parser $wgParser */
+		global $wgParser;
+
+		$wikitext = $wgParser->getSection( $articleAsWikiText, $section );
+		$html = $wgParser->parse( $wikitext, $title, $wgParser->mOptions )->mText;
```

Comment:
It's running the hooks but there's indexing issues with the media array :(. Thanks for bringing this up. I'll fix it.